File: app/src/main/kotlin/jp/cordea/conviviumcalculator/InputDataActivity.kt
package jp.cordea.conviviumcalculator

import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.text.SpannableStringBuilder
import android.widget.EditText
import butterknife.bindView
import io.realm.Realm
import rx.Observable

class InputDataActivity : AppCompatActivity() {

    private val editText: EditText by bindView(R.id.edittext)
    private val fab: FloatingActionButton by bindView(R.id.fab)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_input_data)
        val toolbar = findViewById(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)

        editText.text = SpannableStringBuilder(objToString())
        fab.setOnClickListener {
            val text = editText.text.toString()
            Realm.getInstance(this).let {
                it.beginTransaction()
                it.allObjects(ListItem::class.java).clear()
                if (!text.isEmpty()) {
                    for (arr in stringToObj(text)) {
                        if (arr.size < 2) {
                            continue
                        }
                        var item = it.createObject(ListItem::class.java)

                        item.name = arr[0]
                        item.price = arr[1].toInt()
                        item.switch = if (arr.size > 2) arr[2].toBoolean() else false
                    }
                }
                it.commitTransaction()
                it.close()
                finish()
            }
        }
    }

    private fun stringToObj(csv: String): Array<Array<String>> {
        return Observable
                .just(csv)
                .flatMap { Observable.from(it.split('\n')) }
                .map { it.split(',').toTypedArray() }
                .toList()
                .toBlocking()
                .first()
                .toTypedArray();
    }

    private fun objToString(): String {
        Realm.getInstance(this).let {
            return Observable
                    .from(it.allObjects(ListItem::class.java))
                    .map { "%s,%d,%s".format(it.name, it.price, it.switch.toString()) }
                    .toList()
                    .toBlocking()
                    .first()
                    .joinToString("\n")
        }
    }
}

File: xamarin.droid (fsharp)/ConviviumCalculator/InputDataActivity.fs

namespace ConviviumCalculator

open System
open System.Collections.Generic
open System.Linq
open System.Text

open Android.App
open Android.Content
open Android.OS
open Android.Runtime
open Android.Views
open Android.Widget
open Android.Util
open Android.Support.V7.Widget
open Android.Support.V7.App

open Mono.Data.Sqlite

[<Activity (Label = "InputData", Theme = "@style/AppTheme.NoActionBar")>]
type InputDataActivity() =
  inherit AppCompatActivity()
  
  [<DefaultValue>] val mutable editText: EditText
  [<DefaultValue>] val mutable dbcon: SqliteConnection
  
  member me.InsertData(text: string) =
      async {
          let dbcmd = me.dbcon.CreateCommand()
          let cmd str =
              dbcmd.CommandText <- str
              dbcmd.ExecuteReader() |> ignore
              dbcmd.Dispose()
              
          let trs = me.dbcon.BeginTransaction()
          text.Split([|"\n"|], System.StringSplitOptions.None)
                |> Array.toList
                |> List.map(fun k -> k.Split([|","|], System.StringSplitOptions.None) |> Array.toList)
                |> List.filter(fun k -> k.Length.Equals 3)
                |> List.map(fun k -> new Model.Item(List.nth k 0, int16(List.nth k 1), Convert.ToBoolean(List.nth k 2)))
                |> List.map(fun k -> String.Format("INSERT INTO Convivium (name, price, isSwitch) VALUES ('{0}', {1}, '{2}')", k.name, k.price, k.isSwitch))
                |> List.iter(fun k -> cmd(k))
          trs.Commit()
      }
  
  override me.OnCreate(bundle) =
    base.OnCreate (bundle)
    me.SetContentView (Resource_Layout.InputData)
     
    let toolbar = me.FindViewById<Toolbar>(Resource_Id.toolbar)
    me.SetSupportActionBar toolbar
       
    me.editText <- me.FindViewById<EditText> Resource_Id.edit_text

    me.dbcon <- new SqliteConnection(Constraints.General.connectionString)
    me.dbcon.Open()
    let dbcmd = me.dbcon.CreateCommand()
    dbcmd.CommandText <- "SELECT * FROM Convivium"
    let reader = dbcmd.ExecuteReader()
    me.editText.Text <- [ while reader.Read() do yield String.Format("{0},{1},{2}\n", reader.GetString 0, reader.GetInt16 1, reader.GetString 2) ]
                            |> String.Concat
    dbcmd.Dispose()
        
    let trs = me.dbcon.BeginTransaction()
    let dbcmd = me.dbcon.CreateCommand()
    dbcmd.CommandText <- "DELETE FROM Convivium"
    dbcmd.ExecuteReader() |> ignore
    dbcmd.Dispose()
    trs.Commit()
    
  override me.OnPause() =
      base.OnPause()
      let text: string = me.editText.Text.ToString()
         
      me.InsertData(text) |> Async.RunSynchronously
